### PR TITLE
refactor: remove _incorrectWorkingDirectory

### PR DESCRIPTION
**Pull request by BlackHole1**
Fixed: https://github.com/pnpm/pnpm/issues/3709

Notes:
Do we really need this feature?
[npm/npm-lifecycle](https://github.com/npm/npm-lifecycle) is no longer maintained and has been replaced by [npm/run-script](https://github.com/npm/run-script)
There is no relevant check in `@npmcli/run-script`

### CODE REVIEW

**Comment by zkochan** [comment]
ok, I guess we can change it in pnpm v7

File: index.js [hunk 1]
```diff
@@ -115,6 +116,9 @@ function lifecycle (pkg, stage, wd, opts) {
 }
 
 function _incorrectWorkingDirectory (wd, pkg) {
+  if (process.env.CI || process.env.CODESPACES || isDocker()) {
```
Comment (zkochan) [inline index.js:119]: [is-ci](https://www.npmjs.com/package/is-ci) will detect more CI environments.

File: lib/isDocker.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+// Source Code from: https://github.com/sindresorhus/is-docker/blob/main/index.js
```
Comment (zkochan) [inline lib/isDocker.js:3]: why don't you add it as a dependency?

Comment (BlackHole1) [reply lib/isDocker.js:3]: Because `is-docker` library is `esm`.
As a dependency, an error will be reported

Comment (zkochan) [reply lib/isDocker.js:3]: install v2.2.1
it was commonjs and has the same code.

**Comment by BlackHole1** [comment]
> ok, I guess we can change it in pnpm v7

 Do you mean "Merge this PR. pnpm@v7 remove `_incorrectWorkingDirectory` check"?

**Comment by zkochan** [comment]
We can merge this change to fix it in v6

for v7, we can make a breaking change and remove the code, as you suggested.

**Comment by zkochan** [comment]
or actually, I am not sure. Maybe just remove it and fix it for pnpm v7.
It works in pnpm v6. `unsafe-perm` should be used.

**Comment by BlackHole1** [comment]
> or actually, I am not sure. Maybe just remove it and fix it for pnpm v7. It works in pnpm v6. `unsafe-perm` should be used.

I agree with that. Let's remove `_incorrectWorkingDirectory` from this PR and merge it when we are about to release pnpm v7.

**Pull request merged**